src/services/tools.service.js

```diff
@@ -9,10 +12,17 @@ export default class ToolsService {
   constructor(toolsRepository, log) {
     this.toolsRepository = toolsRepository;
     this.log = log;
+    self = this;
```

Comment:
what is the reasoning for doing this?

Reply:
Why don't you just do this.toolsRepository?